dev/NavigationView/NavigationViewItemPresenter.cpp

```diff
@@ -110,3 +121,18 @@ void NavigationViewItemPresenter::UpdateMargin()
         grid.Margin({ m_leftIndentation, oldGridMargin.Top, oldGridMargin.Right, oldGridMargin.Bottom });
     }
 }
+
+
+void NavigationViewItemPresenter::UpdateCompactPaneLength(double compactPaneLength, bool shouldUpdate)
```

Comment:
1. Having that as a local field is faster then going the DP route.
2. We need that property because there are cases where we just need to update that value but we do not want to update our layout.

Also, I think there was some problem with getting the DP in some cases, but don't quote me on that.

Reply:
Fair enough. Nothing important anyway.